File: server/internal/types/event/events.go
package event

const (
	SYSTEM_INIT       = "system/init"
	SYSTEM_DISCONNECT = "system/disconnect"
	SYSTEM_ERROR      = "system/error"
)

const (
	SIGNAL_OFFER     = "signal/offer"
	SIGNAL_ANSWER    = "signal/answer"
	SIGNAL_PROVIDE   = "signal/provide"
	SIGNAL_CANDIDATE = "signal/candidate"
)

const (
	MEMBER_LIST         = "member/list"
	MEMBER_CONNECTED    = "member/connected"
	MEMBER_DISCONNECTED = "member/disconnected"
)

const (
	CONTROL_LOCKED     = "control/locked"
	CONTROL_RELEASE    = "control/release"
	CONTROL_REQUEST    = "control/request"
	CONTROL_REQUESTING = "control/requesting"
	CONTROL_GIVE       = "control/give"
	CONTROL_CLIPBOARD  = "control/clipboard"
	CONTROL_KEYBOARD   = "control/keyboard"
)

const (
	CHAT_MESSAGE = "chat/message"
	CHAT_EMOTE   = "chat/emote"
)

const (
	FILETRANSFER_LIST    = "filetransfer/list"
	FILETRANSFER_REFRESH = "filetransfer/refresh"
)

const (
	SCREEN_CONFIGURATIONS = "screen/configurations"
	SCREEN_RESOLUTION     = "screen/resolution"
	SCREEN_SET            = "screen/set"
)

const (
	BORADCAST_STATUS  = "broadcast/status"
	BORADCAST_CREATE  = "broadcast/create"
	BORADCAST_DESTROY = "broadcast/destroy"
)

const (
	ADMIN_BAN     = "admin/ban"
	ADMIN_KICK    = "admin/kick"
	ADMIN_LOCK    = "admin/lock"
	ADMIN_MUTE    = "admin/mute"
	ADMIN_UNLOCK  = "admin/unlock"
	ADMIN_UNMUTE  = "admin/unmute"
	ADMIN_CONTROL = "admin/control"
	ADMIN_RELEASE = "admin/release"
	ADMIN_GIVE    = "admin/give"
)

File: src/types/events.ts
export const EVENT = {
  SYSTEM: {
    DISCONNECT: 'system/disconnect',
  },
  SIGNAL: {
    ANSWER: 'signal/answer',
    PROVIDE: 'signal/provide',
  },
  MEMBER: {
    LIST: 'member/list',
    CONNECTED: 'member/connected',
    DISCONNECTED: 'member/disconnected',
  },
  CONTROL: {
    LOCKED: 'control/locked',
    RELEASE: 'control/release',
    REQUEST: 'control/request',
    REQUESTING: 'control/requesting',
    GIVE: 'control/give',
    CLIPBOARD: 'control/clipboard',
    KEYBOARD: 'control/keyboard',
  },
  SCREEN: {
    CONFIGURATIONS: 'screen/configurations',
    RESOLUTION: 'screen/resolution',
    SET: 'screen/set',
  },
  BROADCAST: {
    STATUS: 'broadcast/status',
    CREATE: 'broadcast/create',
    DESTROY: 'broadcast/destroy',
  },
  ADMIN: {
    BAN: 'admin/ban',
    KICK: 'admin/kick',
    MUTE: 'admin/mute',
    UNMUTE: 'admin/unmute',
    LOCK: 'admin/lock',
    UNLOCK: 'admin/unlock',
    CONTROL: 'admin/control',
    RELEASE: 'admin/release',
    GIVE: 'admin/give',
  },
} as const

export type Events = typeof EVENT

export type WebSocketEvents =
  | SystemEvents
  | SignalEvents
  | MemberEvents
  | ControlEvents
  | ScreenEvents
  | BroadcastEvents
  | AdminEvents

export type SystemEvents = typeof EVENT.SYSTEM.DISCONNECT

export type SignalEvents = typeof EVENT.SIGNAL.ANSWER | typeof EVENT.SIGNAL.PROVIDE

export type MemberEvents = typeof EVENT.MEMBER.LIST | typeof EVENT.MEMBER.CONNECTED | typeof EVENT.MEMBER.DISCONNECTED

export type ControlEvents =
  | typeof EVENT.CONTROL.LOCKED
  | typeof EVENT.CONTROL.RELEASE
  | typeof EVENT.CONTROL.REQUEST
  | typeof EVENT.CONTROL.GIVE
  | typeof EVENT.CONTROL.CLIPBOARD
  | typeof EVENT.CONTROL.KEYBOARD

export type ScreenEvents = typeof EVENT.SCREEN.CONFIGURATIONS | typeof EVENT.SCREEN.RESOLUTION | typeof EVENT.SCREEN.SET

export type BroadcastEvents =
  | typeof EVENT.BROADCAST.STATUS
  | typeof EVENT.BROADCAST.CREATE
  | typeof EVENT.BROADCAST.DESTROY

export type AdminEvents =
  | typeof EVENT.ADMIN.BAN
  | typeof EVENT.ADMIN.KICK
  | typeof EVENT.ADMIN.LOCK
  | typeof EVENT.ADMIN.UNLOCK
  | typeof EVENT.ADMIN.MUTE
  | typeof EVENT.ADMIN.UNMUTE
  | typeof EVENT.ADMIN.CONTROL
  | typeof EVENT.ADMIN.RELEASE
  | typeof EVENT.ADMIN.GIVE
